lib/cocoapods-core/dependency.rb

```diff
@@ -95,8 +95,7 @@ def initialize(name = nil, *requirements)
             raise Informative, 'A dependency with a specified podspec repo may ' \
               "not include other source parameters (#{name})."
           end
-        else
-          @external_source = additional_params
+        elsif @external_source = additional_params
           unless requirements.empty?
             raise Informative, 'A dependency with an external source may not ' \
```

Comment:
I don't understand when/why we would expect dependencies with an external source to not specify a version requirement.

Reply:
agreed, but I dont think this is the place to change that